test-network-function/networking/suite.go

```diff
@@ -80,26 +80,31 @@ func testDefaultNetworkConnectivity(env *config.TestEnvironment, count int) {
 				ginkgo.Skip("Orchestrator is not deployed, skip this test")
 			}
 			found := false
+			containerping := false
+
 			for _, cut := range env.ContainersUnderTest {
 				if _, ok := env.ContainersToExcludeFromConnectivityTests[cut.ContainerIdentifier]; ok {
 					tnf.ClaimFilePrintf("Skipping container %s because it is excluded from connectivity tests (default)", cut.ContainerConfiguration.PodName)
 
 					continue
 				}
-				found = true
-				context := cut.Oc
-				testOrchestrator := env.TestOrchestrator
-				ginkgo.By(fmt.Sprintf("a Ping is issued from %s(%s) to %s(%s) %s", testOrchestrator.Oc.GetPodName(),
-					testOrchestrator.Oc.GetPodContainerName(), cut.Oc.GetPodName(), cut.Oc.GetPodContainerName(),
-					cut.DefaultNetworkIPAddress))
-				testPing(testOrchestrator.Oc, cut.DefaultNetworkIPAddress, count)
-				ginkgo.By(fmt.Sprintf("a Ping is issued from %s(%s) to %s(%s) %s", cut.Oc.GetPodName(),
-					cut.Oc.GetPodContainerName(), testOrchestrator.Oc.GetPodName(), testOrchestrator.Oc.GetPodContainerName(),
-					testOrchestrator.DefaultNetworkIPAddress))
-				testPing(context, testOrchestrator.DefaultNetworkIPAddress, count)
+				if cut.ContainerIdentifier.HasPing {
+					containerping = true
+					found = true
+					context := cut.Oc
+					testOrchestrator := env.TestOrchestrator
+					ginkgo.By(fmt.Sprintf("a Ping is issued from %s(%s) to %s(%s) %s", testOrchestrator.Oc.GetPodName(),
+						testOrchestrator.Oc.GetPodContainerName(), cut.Oc.GetPodName(), cut.Oc.GetPodContainerName(),
+						cut.DefaultNetworkIPAddress))
+					testPing(testOrchestrator.Oc, cut.DefaultNetworkIPAddress, count)
+					ginkgo.By(fmt.Sprintf("a Ping is issued from %s(%s) to %s(%s) %s", cut.Oc.GetPodName(),
+						cut.Oc.GetPodContainerName(), testOrchestrator.Oc.GetPodName(), testOrchestrator.Oc.GetPodContainerName(),
+						testOrchestrator.DefaultNetworkIPAddress))
+					testPing(context, testOrchestrator.DefaultNetworkIPAddress, count)
+				}
 			}
-			if !found {
-				ginkgo.Skip("No container found suitable for connectivity test")
+			if !found || !containerping {
```

Comment:
I'd prefer to have two separate checks here. One for "if !found" and another one for "if !containerping". That would allow us to be specific with the Skip message when debugging.